src/main/scala/mesosphere/marathon/state/AppDefinition.scala

```diff
@@ -526,4 +530,46 @@ object AppDefinition {
       }
     }
   }
+
+  def residentUpdateIsValid(from: AppDefinition): Validator[AppDefinition] = {
+    def changeNoVolumes: Validator[AppDefinition] = new Validator[AppDefinition] {
+      override def apply(to: AppDefinition): Result = {
+        val fromVolumes = from.persistentVolumes
+        val toVolumes = to.persistentVolumes
+        def sameSize = fromVolumes.size == toVolumes.size
+        def noChange = from.persistentVolumes.forall { fromVolume =>
+          toVolumes.find(_.containerPath == fromVolume.containerPath).contains(fromVolume)
```

Comment:
please add a check for `persistentVolume.persistent.size` – that's the size of the volume itself which will also need to be reserved and must not change.